mobile/lib/features/trade/trade_bottom_sheet_tab.dart

```diff
@@ -395,4 +397,26 @@ class _TradeBottomSheetTabState extends State<TradeBottomSheetTab>
 
   @override
   bool get wantKeepAlive => true;
+
+  /// Returns the trade action depending on the trade values and if a channel exists
+  TradeAction getTradeAction(TradeValues tradeValues, bool hasChannel) {
+    if (!hasChannel) {
+      return TradeAction.openChannel;
+    }
+
+    if (tradeValues.openQuantity == tradeValues.contracts) {
```

Comment:
I feel like I'm missing something, how does it know the direction?

Reply:
The open quantity is only set for the opposite direction of the position. so if the user buys/sells as many contracts as they currently have open, the position will be closed.